alpine/ExamplesWithoutPicManager/LandauDampingParameterList.cpp

```diff
@@ -0,0 +1,425 @@
+
+        int arg = 1;
+
+        Vector_t<int, Dim> nr;
+        for (unsigned d = 0; d < Dim; d++) {
+            nr[d] = std::atoi(argv[arg++]);
+        }
+
+        static IpplTimings::TimerRef mainTimer        = IpplTimings::getTimer("total");
+        static IpplTimings::TimerRef particleCreation = IpplTimings::getTimer("particlesCreation");
+        static IpplTimings::TimerRef dumpDataTimer    = IpplTimings::getTimer("dumpData");
+        static IpplTimings::TimerRef PTimer           = IpplTimings::getTimer("pushVelocity");
+        static IpplTimings::TimerRef RTimer           = IpplTimings::getTimer("pushPosition");
+        static IpplTimings::TimerRef updateTimer      = IpplTimings::getTimer("update");
+        static IpplTimings::TimerRef DummySolveTimer  = IpplTimings::getTimer("solveWarmup");
+        static IpplTimings::TimerRef SolveTimer       = IpplTimings::getTimer("solve");
+        static IpplTimings::TimerRef domainDecomposition = IpplTimings::getTimer("loadBalance");
+
+        IpplTimings::startTimer(mainTimer);
+
+        const size_type totalP = std::atoll(argv[arg++]);
+        const unsigned int nt  = std::atoi(argv[arg++]);
+
+        msg << "Landau damping" << endl
+            << "nt " << nt << " Np= " << totalP << " grid = " << nr << endl;
+
+        using bunch_type = ChargedParticles<PLayout_t<double, Dim>, double, Dim>;
+
+        std::unique_ptr<bunch_type> P;
+
+        ippl::NDIndex<Dim> domain;
+        for (unsigned i = 0; i < Dim; i++) {
+            domain[i] = ippl::Index(nr[i]);
+        }
+
+        std::array<bool, Dim> isParallel;
+        isParallel.fill(true);
+
+        // create mesh and layout objects for this problem domain
+        Vector_t<double, Dim> kw = 0.5;
+        double alpha             = 0.05;
+        Vector_t<double, Dim> rmin(0.0);
+        Vector_t<double, Dim> rmax = 2 * pi / kw;
+
+        Vector_t<double, Dim> hr = rmax / nr;
+        // Q = -\int\int f dx dv
+        double Q = std::reduce(rmax.begin(), rmax.end(), -1., std::multiplies<double>());
+        Vector_t<double, Dim> origin = rmin;
+        const double dt              = std::min(.05, 0.5 * *std::min_element(hr.begin(), hr.end()));
+
+        const bool isAllPeriodic = true;
+        Mesh_t<Dim> mesh(domain, hr, origin);
+        FieldLayout_t<Dim> FL(MPI_COMM_WORLD, domain, isParallel, isAllPeriodic);
+        PLayout_t<double, Dim> PL(FL, mesh);
+
+        std::string solver = argv[arg++];
+        // MOD: Setup of Preconditioner Start
+        int gauss_seidel_inner_iterations;
+        int gauss_seidel_outer_iterations;
+        int newton_level;
+        int chebyshev_degree;
+        int richardson_iterations;
+        int communication;
+        std::string preconditioner_type = "";
+
+        if (solver == "OPEN") {
+            throw IpplException("LandauDamping",
+                                "Open boundaries solver incompatible with this simulation!");
+        }
+
+        if (solver == "PCG") {
```

Comment:
Should an exception be thrown when the appropriate arguments are not given (when solver type is PCG and according to each preconditioner_type)?

Reply:
I assumed this LandauDamping version was only for testing purposes, therefore I think the responsibilty of using the preconditioners correctly is on the user.

Reply:
What do you think @srikrrish ?

Reply:
Yeah an exception would be nice but we don't do this for many of the other arguments in the min-app so I think it is ok. However, as I explained in the other comment before the options should be explained.